.github/workflows/smoke.yaml

```diff
@@ -4,6 +4,7 @@ on:
   workflow_dispatch:
   schedule:
     - cron: '0 7 25 * *'
+  pull_request:
```

Comment:
What's the reasoning here? I think these are too heavy to do on PRs, and super unlikely to break. Monthly seems fine. Or if we are going to do them on PR then I think we should remove the scheduled run.